### PR TITLE
Update HuggingFace server dependencies versions

### PR DESCRIPTION
**Pull request by AyushSawant18588**
**What this PR does / why we need it**:
Updated to use vLLM version 0.6.4.post1, and upgraded setuptools and pip in the Hugging Face server Docker image. These updates address security vulnerabilities identified in the KServe controller and Hugging Face server images.

Solves this issue: https://github.com/kserve/kserve/issues/4076

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- Completed sanity checks by running an inference service using the Hugging Face vLLM Openvino Runtime image and Hugging Face vLLM GPU Runtime image

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -115,15 +115,15 @@ require (
 	go.opentelemetry.io/otel/trace v1.29.0 // indirect
 	go.uber.org/atomic v1.11.0 // indirect
 	go.uber.org/multierr v1.11.0 // indirect
-	golang.org/x/crypto v0.26.0 // indirect
+	golang.org/x/crypto v0.31.0 // indirect
```
Comment (yuzisun) [inline go.mod]: We should separate go mod update to a different PR, it is easier to cherry pick if we need to back patch to easier releases.

Comment (AyushSawant18588) [reply go.mod]: okay I will remove these changes from this PR
